### PR TITLE
feat(to-have-value): added new rule prefer-to-have-value

### PR DESCRIPTION
**Pull request by benmonro**
**What**: Add new rule prefer-to-have-value

**Why**: all the matchers!

**How**: with code.

**Checklist**:

- [ ] Documentation
- [ ] Tests
- [ ] Ready to be merged

### CODE REVIEW

File: src/rules/prefer-to-have-value.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+export const meta = {
+  docs: {
+    category: "jest-dom",
+    description: "prefer toHaveValue over checking  element.value",
+    url: "prefer-to-have-value",
+    recommended: false,
```
Comment (nickmccurdy) [inline src/rules/prefer-to-have-value.js:15]: Why not recommend this rule? Seems safe and it's still fixible.

Comment (benmonro) [reply src/rules/prefer-to-have-value.js:15]: Wanted to vet it a bit with the new smoke test feature we added. But yeah this is the plan

File: src/rules/prefer-to-have-value.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  fixable: "code",
+  messages: {
+    "use-to-have-value": `Prefer .toBeInTheDocument() for asserting DOM node existence`,
+  },
+};
+const messageId = "use-to-have-value";
+
+export const create = (context) => ({
+  // expect(element.value).toBe('foo') / toEqual / toStrictEqual
+  [`CallExpression[callee.property.name=/to(Be|(Strict)?Equal)$/][callee.object.arguments.0.property.name=value][arguments.0][callee.object.callee.name=expect]`](
```
Comment (AntonNiklasson) [inline src/rules/prefer-to-have-value.js:26]: I'm not sure the clause `[arguments.0]` is making a difference here. Or is it?

File: src/rules/prefer-to-have-value.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            node.callee.object.object.arguments[0].property.range[1],
+          ]),
+          fixer.replaceText(node.callee.property, "toHaveValue"),
+        ];
+      },
+    });
+  },
+
+  //expect(element).toHaveAttribute('value', 'foo')  / Property
+  [`CallExpression[callee.property.name=/toHave(Attribute|Property)/][arguments.0.value=value][arguments.1][callee.object.callee.name=expect], CallExpression[callee.property.name=/toHave(Attribute|Property)/][arguments.0.value=value][arguments.1][callee.object.object.callee.name=expect][callee.object.property.name=not]`](
```
Comment (AntonNiklasson) [inline src/rules/prefer-to-have-value.js:64]: Is `[arguments.1]` needed here?

File: src/rules/prefer-to-have-value.js (hunk 2 above)
Comment (benmonro) [reply src/rules/prefer-to-have-value.js:26]:
```suggestion
  [`CallExpression[callee.property.name=/to(Be|(Strict)?Equal)$/][callee.object.arguments.0.property.name=value][callee.object.callee.name=expect]`](
```

File: src/rules/prefer-to-have-value.js (hunk 3 above)
Comment (benmonro) [reply src/rules/prefer-to-have-value.js:64]: yeah here's it's needed.  its essentially a non null check.

**Pull request merged**